py3status/modules/google_calendar.py

```diff
@@ -0,0 +1,438 @@
+# -*- coding: utf-8 -*-
+"""
+Display upcoming Google Calendar events.
+
+This module will display information about upcoming Google Calendar events in
+either compact or full format which can be toggled with a button press. The
+events URL may also be opened in a web browser with a button press.
+
+Configuration parameters:
+    auth_token_path: The path to where the access/refresh token will be saved
+        after successful credential authorization.
+        (default '~/.credentials')
+    button_open_url: Opens the URL for the clicked on event in the default
+        web browser.
+        (default 3)
+    button_toggle_format: The button used to toggle between output formats
+        format_event_full and format_event_compact.
+        (default 1)
+    cache_timeout: How often the module is refreshed in seconds
+        (default 60)
+    client_secret_path: the path to your client_secret.json file which
+        contains your OAuth 2.0 credentials.
+        (default '')
+    colors: A list of color strings specifying what color to display events
```

Comment:
This is fancy. You might be interested in `thresholds` + `threshold_get_color`... as it does what you're doing here and it also support gradient should the users want them... so the rest of the events does not have to be painted white! 🌈

Reply:
I'm a bit confused as to how to use thresholds here. Can we define our own constants or do we have to use existing ones like `color_good` or `color_degraded`? The colors array there isn't meant to change the color based on any info about the event, it's just there to display individual events in whatever color the user wants. Maybe I'm not understanding what you mean?

Reply:
For event thresholds, we could use general thresholds 
```
thresholds = [(1, '#0FF'), (2, '#00F'), (3, '#0F0'), (4, '#FF0'), (5, '#FFF')]
                 Cyan         Blue        Green        Yellow        Red
```
... with starting `\?color=thresholds {event_name}`..
```
    format_event_1 = '\?color=thresholds {summary} {time_to}'
    format_event_2 = '\?color=thresholds {summary} ({start_time} - {end_time}, {start_date})'
```
This would print `5` different event colors. Now, what if you have `20` events? This acts same as your current setup. Enable `gradients = True` and you have gradients colors spreading out to all `20` events.

>do we have to use existing ones like color_good or color_degraded

No. You don't have to. See above. If you want it, it usually looks like this.
```
thresholds = [(1, 'bad'), (2, 'degraded'), (3, 'good')]
Event 1 bad. Event 2 degraded. Event 3 good
```
```
thresholds = [(1, 'bad'), (4, 'degraded'), (3, 'good')]
Event 1, 2, 3 bad. Event 4 degraded, etc...
```

`format_event_1` and `format_event_2` looks bad.
Maybe `format_event` and `format_event_full` ?

Reply:
Ahh, I see, that's a great explanation. Thanks. 

"Maybe format_event and format_event_full ?"

Okay, that sounds good.